topics-tools/gen-topics-database/main.go

```diff
@@ -4422,10 +4454,20 @@ func populateTopicAndContentStructs(topics []TopicResponseStore, content []Conte
 					if topics[parentID].Next.SubtopicIds == nil {
 						topics[parentID].Next.SubtopicIds = &[]string{idAndName}
 					} else {
-						*topics[parentID].Next.SubtopicIds = append(*topics[parentID].Next.SubtopicIds, idAndName)
+						if !stringInSlice(idAndName, *topics[parentID].Next.SubtopicIds) {
+							*topics[parentID].Next.SubtopicIds = append(*topics[parentID].Next.SubtopicIds, idAndName)
+						}
+					}
+
+					if pageType[id] != pageTopicAndContent {
+						if pageType[id] == pageTopic {
+							// add content to topic
+							pageType[id] = pageTopicAndContent
+						} else {
+							pageType[id] = pageContent
+						}
```

Comment:
These appear to be duplicates of code written previously (line 4381 & 4354). Is this something that can be abstracted out into a function and reused?

Reply:
No, the inner part of the iff is different, they are ok.